modules/tinymce/src/plugins/codesample/test/ts/webdriver/CodeSampleCopyAndPasteTest.ts

```diff
@@ -0,0 +1,82 @@
+import { Keys, RealClipboard } from '@ephox/agar';
+import { beforeEach, describe, it } from '@ephox/bedrock-client';
+import { PlatformDetection } from '@ephox/sand';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/codesample/Plugin';
+
+import * as TestUtils from '../module/CodeSampleTestUtils';
+
+describe('webdriver.tinymce.plugins.codesample.CodeSampleCopyAndPasteTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'codesample',
+    toolbar: 'codesample',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ]);
+
+  const pressEnter = (editor: Editor) => {
+    const overrides = { keyCode: Keys.enter() };
+
+    const getInputEventArgs = (eventType: string) => ({ ...new window.InputEvent(eventType), ...overrides });
+    const getKeyboardEventArgs = (eventType: string) => ({ ...new window.KeyboardEvent(eventType), ...overrides });
+
+    editor.dispatch('keydown', getKeyboardEventArgs('keydown'));
+    editor.dispatch('beforeinput', getInputEventArgs('beforeinput'));
+    editor.dispatch('input', getInputEventArgs('input'));
+    editor.dispatch('keyup', getKeyboardEventArgs('keyup'));
```

Comment:
You pretty much should never be doing this, especially in a webdriver test. In a webdriver test we should be using `RealKeys` and elsewhere we should be using `Keyboard` to do that.

The only time we don't use them is if we need something else to manually happen between the keyboard events (in Permanent Pen for example Agar doesn't fire the input event, though that is something we need to add to Agar).

Reply:
done

Reply:
@lorenzo-pomili sorry I missed something in the above... the other thing we should use is the McAgar helper functions. So we could use `TinyContentActions.keystroke(editor, Keys.enter());` here.

Also as a side note, if it's not for TinyMCE, there's the `Keyboard.activeKeystroke` API as well instead of manually getting the focused element and passing it.

Reply:
ok, I changed it, thanks